src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +47,1075 @@ class Outer
+            }
+        }
+
+        [Fact]
+        public void Duplication_03()
+        {
+            var source1 = """
+                using System;
+
+                partial class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(1);
+                    }
+                }
+                """;
+
+            var source2 = """
+                partial class C
+                {
+                }
+                """;
+
+            var main = """
+                using System;
+
+                file class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(2);
+                    }
+                }
+
+                class Program
+                {
+                    static void Main()
+                    {
+                        C.M();
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(new[] { source1, source2, main }); // expectedOutput: 2
+            comp.VerifyDiagnostics();
+
+            var cs = comp.GetMembers("C");
+            Assert.Equal(2, cs.Length);
+
+            var c0 = cs[0];
+            Assert.True(c0 is SourceMemberContainerTypeSymbol { IsFile: false });
+
+            var syntaxReferences = c0.DeclaringSyntaxReferences;
+            Assert.Equal(2, syntaxReferences.Length);
+            Assert.Equal(comp.SyntaxTrees[0], syntaxReferences[0].SyntaxTree);
+            Assert.Equal(comp.SyntaxTrees[1], syntaxReferences[1].SyntaxTree);
+
+            var c1 = cs[1];
+            Assert.True(c1 is SourceMemberContainerTypeSymbol { IsFile: true });
+            Assert.Equal(comp.SyntaxTrees[2], c1.DeclaringSyntaxReferences.Single().SyntaxTree);
+        }
+
+        [Fact]
+        public void Duplication_04()
+        {
+            var source1 = """
+                using System;
+
+                class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(1);
+                    }
+                }
+                """;
+
+            var main = """
+                using System;
+
+                file partial class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(Number);
+                    }
+                }
+
+                file partial class C
+                {
+                    private static int Number => 2;
+                }
+
+                class Program
+                {
+                    static void Main()
+                    {
+                        C.M();
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(new[] { source1, main }); // expectedOutput: 2
+            comp.VerifyDiagnostics();
+
+            var cs = comp.GetMembers("C");
```

Comment:
I'm worried about a slight usability problem for some APIs. This may also show up in the IDE (navigation bar would show two types "C" in this file, which we'll need to disambiguate somehow)

Reply:
I think it's probably a good thing to show file types in the IDE navigation bar, even if it means that multiple types with the same name may show up. IDE may want to consider adjusting the presentation of the types to disambiguate. That's their decision to make though.